internal/instrumentation/sdk.go

```diff
@@ -232,6 +235,56 @@ func (i *sdkInjector) setInitContainerSecurityContext(pod corev1.Pod, securityCo
 	return pod
 }
 
+func (i *sdkInjector) injectImagePullSecrets(insts languageInstrumentations, pod corev1.Pod) corev1.Pod {
+	// Collect imagePullSecrets from all instrumentation instances
+	var imagePullSecrets []corev1.LocalObjectReference
+
+	// Check each instrumentation type for imagePullSecrets
+	if insts.Java.Instrumentation != nil && len(insts.Java.Instrumentation.Spec.ImagePullSecrets) > 0 {
```

Comment:
Is there any reason to not `insts.Java.Instrumentation.Spec.ImagePullSecrets != ""`?